x-pack/plugins/security_solution/public/common/utils/route/spy_routes.tsx

```diff
@@ -40,17 +41,24 @@ export const SpyRouteComponent = memo<
     useEffect(() => {
       if (pageName && !deepEqual(route.pathName, pathname)) {
         if (isInitializing && detailName == null) {
+          const routeAction = {
```

Comment:
@stephmilovic,  @patrykkopycinski 
would really like your feedback/review here. Both of you seem to have been the last ones to touch this code. It took me a while to get to this point and identify the problem and the fix below ensure that it only applies to the Administration pages, but I wonder if others also might have this problem. 

I think it is a problem for us because because we don't have a `detailName` for our List pages.

Also - FYI: I noticed that the reducer that handles the this action of `updateRouteWithOutSearch` does not actually "remove" the `search` before merging in this new payload (`action.route`), thus why I'm setting it here (for Administration only) to either what we have in `search` or empty string to ensure prior params are removed.

I have another Issue that needs to be address which I will chase next - the react-router `state` is being dropped somehow. My guess is due to a similar issue like this one, but I have not been able to identify it yet (got any ideas 😄 ) . We use react-router route state to carry state from other kibana plugins to our pages and I [recently had to implement a work-around ](https://github.com/elastic/kibana/blob/f340d713771ae160823bb243df20414df0413e7f/x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_page.tsx#L43-L50)to ensure it was not lost.

Reply:
this makes sense to me